client/matching/client.go

```diff
@@ -232,6 +232,33 @@ func (c *clientImpl) ListTaskListPartitions(
 	return client.ListTaskListPartitions(ctx, request, opts...)
 }
 
+func (c *clientImpl) GetTaskListsForDomain(
+	ctx context.Context,
+	request *types.MatchingGetTaskListsForDomainRequest,
+	opts ...yarpc.CallOption,
+) (*types.GetTaskListsForDomainResponse, error) {
+	opts = common.AggregateYarpcOptions(ctx, opts...)
+	clients := c.clients.GetAllTaskLists()
```

Comment:
The result could be large?

Reply:
Should I use pagination here? What do you think?

Reply:
Leave a comment in GetAllClients()